pandas/compat/__init__.py

```diff
@@ -36,7 +39,7 @@
     from pandas._typing import F
 
 
-def set_function_name(f: F, name: str, cls) -> F:
+def set_function_name(f: F, name: str, cls: Any) -> F:
```

Comment:
`type` could work for `cls` (but not sure whether all call sites use a type object).